nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/impl/adapters/AbstractTypeAdapter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * (C) Copyright 2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ *
+ */
+
+package org.nuxeo.ecm.automation.core.impl.adapters;
+
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.TypeAdaptException;
+import org.nuxeo.ecm.automation.TypeAdapter;
+import org.nuxeo.ecm.automation.core.scripting.Expression;
+import org.nuxeo.ecm.automation.core.scripting.Scripting;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.api.PathRef;
+
+/**
+ * @since 9.1
+ */
+public abstract class AbstractTypeAdapter implements TypeAdapter {
+
+    protected DocumentRef createRef(OperationContext ctx, String value) throws TypeAdaptException {
```

Comment:
Actually make it `public static` (as we need to call it from `OperationChainContribution`) and include a Javadoc explaining the different use cases for `value`. As per my email we should explain the case of starting with `"."` and actually change the code to allow exactly `"."` or `startWith("./")`. Finally, would you mind adding a unit test for that code path as today it's not tested at all. Ask me for details.